src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ImmutableByteArrayInterop.cs

```diff
@@ -54,14 +49,12 @@ internal static unsafe class ImmutableByteArrayInterop
         /// whose underlying backing field is modified.
         /// </remarks>
         /// <returns>An immutable array.</returns>
-        internal static ImmutableArray<byte> DangerousCreateFromUnderlyingArray(ref byte[]? array)
+        internal static ImmutableArray<byte> DangerousCreateFromUnderlyingArray([DisallowNull] ref byte[]? array)
```

Comment:
The error occurs because `MethodBodyBlock.GetILContent` returns `default(ImmutableArray<byte>)` if the method does not have a method body.

Returning `default(ImmutableArray<T>)` feels weird because even quering for its length fails with an exception, you first need to call `IsDefault`. But unless you want to return an empty immutable array (guess not for compatibility reasons), I will revert it.

Reply:
Fixed.